src/libraries/System.IO.Pipes/tests/PipeStreamConformanceTests.cs

```diff
@@ -278,7 +278,7 @@ public async Task CancelTokenOn_ServerWaitForConnectionAsyncWithOuterCancellatio
             Task waitForConnectionTask = server.WaitForConnectionAsync(cts.Token);
 
             Assert.True(InteropTest.CancelIoEx(server.SafePipeHandle), "Outer cancellation failed");
-            await Assert.ThrowsAsync<IOException>(() => waitForConnectionTask);
+            await Assert.ThrowsAsync<OperationCanceledException>(() => waitForConnectionTask);
```

Comment:
in theory this is a breaking change. I wanted to align `WaitForConnectionAsync` with `ReadAsync` and `WriteAsync`.

Reply:
> in theory this is a breaking change

We've gone to some lengths to maintain this behavior from .NET Framework.  If we want to take a breaking change around it, I'm ok with that, but I don't have a great sense of how impactful that will be.

Reply:
> I don't have a great sense of how impactful that will be